### PR TITLE
add dockerfile and components

### PR DESCRIPTION
**Pull request by madebyTimo**
This PR adds the components for our sharry fileserver image.

To test run `docker compose up --build` and test if the fileserver is accessible via http://localhost/fileserver/ and the login via keycloak works.

Please use the corresponding keycloak test-data for testing.

### CODE REVIEW

**Comment by madebyTimo** [comment]
The tag for the setup container with the corresponding test data is `feature-update-keycloak-sharry`

Please also review the corresponding test-data pr https://github.com/Gamify-IT/test-data/pull/22

File: sharry.conf [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    alias-member-enabled = true
+
+    auth {
+      internal {
+        enabled = false
+      }
+      oauth = [
+        {
+          enabled = true
+          id = "gameify-it"
```
Comment (programminghoch10) [inline sharry.conf]:
```suggestion
          id = "gamify-it"
```

File: .github/workflows/deploy-docker-image-on-release.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
```
Comment (programminghoch10) [inline .github/workflows/deploy-docker-image-on-release.yml]:
```suggestion
        uses: docker/login-action@v2
```

File: .github/workflows/deploy-docker-image-on-release.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
```
Comment (programminghoch10) [inline .github/workflows/deploy-docker-image-on-release.yml]:
```suggestion
        uses: docker/metadata-action@v4
```

File: .github/workflows/deploy-docker-image-on-release.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
```
Comment (programminghoch10) [inline .github/workflows/deploy-docker-image-on-release.yml]:
```suggestion
        uses: docker/build-push-action@v3
```

**Review by programminghoch10 — CHANGES_REQUESTED** [review]

**Comment by Aaron2000119** [comment]
While running it on windows, i get this error `': No such file or directory't execute 'bash` in the fileserver container.

**Comment by madebyTimo** [comment]
@Aaron2000119 on Linux there is no such problem, can you provide the full logs?

**Comment by Aaron2000119** [comment]
': No such file or directoryn't execute 'bash
': No such file or directoryn't execute 'bash
': No such file or directoryn't execute 'bash
': No such file or directoryn't execute 'bash
': No such file or directoryn't execute 'bash
': No such file or directoryn't execute 'bash
': No such file or directoryn't execute 'bash
': No such file or directoryn't execute 'bash
': No such file or directoryn't execute 'bash
fileserver exited with code 127

**Comment by madebyTimo** [comment]
We solved the problem. It was caused by wrong line-endings

File: docker-compose.yaml [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+  # reverse-proxy
+  reverse-proxy:
+    container_name: reverse-proxy
+    image: nginx
+    restart: always
+    volumes:
+      - ./.nginx/compose/nginx.conf:/etc/nginx/conf.d/default.conf:ro
+    ports:
+      - "80:80"
```
Comment (programminghoch10) [inline docker-compose.yaml]: Newline at the end makes the file look cleaner.
```suggestion
      - "80:80"

```

File: entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    --expression "s|###KEYCLOAK_CLIENT_ID###|${KEYCLOAK_CLIENT_ID}|g" \
+    --expression "s|###KEYCLOAK_CLIENT_SECRET###|${KEYCLOAK_CLIENT_SECRET}|g" \
+    --expression "s|###KEYCLOAK_ISSUER###|${KEYCLOAK_ISSUER}|g" \
+    --expression "s|###KEYCLOAK_URL###|${KEYCLOAK_URL}|g" \
+    --expression "s|###POSTGRES_USER###|${POSTGRES_USER}|g" \
+    --expression "s|###POSTGRES_PASSWORD###|${POSTGRES_PASSWORD}|g" \
+    --expression "s|###POSTGRES_URL###|${POSTGRES_URL}|g" \
+    "/opt/sharry/conf/sharry.conf"
+
+exec "$@"
```
Comment (programminghoch10) [inline entrypoint.sh]:
```suggestion
exec "$@"

```

**Review by programminghoch10 — CHANGES_REQUESTED** [review]

**Review by programminghoch10 — APPROVED** [review]

**Review by Aaron2000119 — APPROVED** [review]

**Pull request merged**